tests/test_xopen.py

```diff
@@ -89,40 +119,72 @@ def fname(request):
     return request.param
 
 
+@pytest.fixture(params=ALL_READERS_WITH_EXTENSION)
+def reader(request):
+    return request.param
+
+
+@pytest.fixture(params=ALL_WRITERS_WITH_EXTENSION)
+def writer(request):
+    return request.param
+
+
+@contextmanager
+def disable_binary(tmp_path, binary_name):
+    try:
+        binary_path = shutil.which(binary_name)
+        if binary_path:
+            shutil.copy(binary_path, str(tmp_path))
+            os.chmod(str(tmp_path / binary_name), 0)
+        path = os.environ["PATH"]
+        os.environ["PATH"] = str(tmp_path)
+        yield
+    finally:
+        os.environ["PATH"] = path
+
+
 @pytest.fixture
 def lacking_pigz_permissions(tmp_path):
     """
     Set PATH to a directory that contains a pigz binary with permissions set to 000.
     If no suitable pigz binary could be found, PATH is set to an empty directory
     """
-    pigz_path = shutil.which("pigz")
-    if pigz_path:
-        shutil.copy(pigz_path, str(tmp_path))
-        os.chmod(str(tmp_path / "pigz"), 0)
+    with disable_binary(tmp_path, "pigz"):
+        yield
+
 
-    path = os.environ["PATH"]
-    os.environ["PATH"] = str(tmp_path)
-    yield
-    os.environ["PATH"] = path
+@pytest.fixture
+def lacking_pbzip2_permissions(tmp_path):
+    """
```

Comment:
Can you please move this docstring to `disable_binary` (and adjust it appropriately)?